### DISCUSSION TITLE
Import item with metadata pulled from external sidecar file instead of the embedded EXIF headers

### DISCUSSION THREAD

**Discussion by Nocifer** [photoprism/photoprism#2675] [Q&A] [answered]
I'm currently in the process of importing my collection into Photoprism. I found out that I have a few videos that were captured with a camera that didn't have the correct time set. These videos are MTS files, so I only have read access to the embedded EXIF metadata. 

The problem is that, while I've now created accompanying XMP files for them that contain the correct date, Photoprism stubbornly insists on importing them based on the embedded metadata. Changing filenames or the XMP metadata does nothing - the files are imported with the wrong date and time, and renamed as such.

So, my questions:

1) Am I doing something wrong? I only created the XMP files after the initial import of these problematic files; I've made sure to delete the YAML sidecars between my attempts, but maybe there is some other residual file I need to edit/delete in order to make Photoprism correctly re-import these files?
2) If this is how it's supposed to be by default, is there some way to tell Photoprism to give priority to external metadata from XMP files, if they exist?

After doing a more thorough "search for similar discussions", I stumbled onto #2331. (I'm leaving the previous content as is for documentation purposes). As far as I gathered, the only way to do what I want for the time being is using a JSON metadata file instead of an XMP one. But for the life of me I can't find a sample of the JSON file format anywhere, and I kind of need one because comparing a JSON file created by `exiftool` and one exported from Google Photos (which is mentioned in the documentation as being compatible for importing with Photoprism) reveals vastly different contents.

So my new question is this: will using a JSON file correctly prompt Photoprism to use the correct metadata included therein instead of the wrong metadata embedded in the file? And if so, is the JSON output produced by `exiftool -json $FILE > $FILE.json` compatible with Photoprism? If not, where can I find a sample JSON file so I can use that as a blueprint?

**Comment by lastzero** [comment, marked as answer]
https://github.com/photoprism/photoprism/tree/develop/internal/meta/testdata

https://photoprism.app/kb/metadata

**Reply by Nocifer** [reply]
So according to these links, if an XMP sidecar is found next to the file it will be used for indexing; and one of the XMP tags supported by Photoprism is the DateCreated tag, so if my XMP sidecar contains that tag then it should be compatible. Now here's my problem:

- I have a file 20090624-00255.mts
- Its embedded EXIF metadata contain the (exiftools's) DateFileOriginal tag with the timestamp "2007:01:01 00:00:26+00:00"
- I create (via Digikam) a file 20090426-00255.xmp (or 20090426-00255.mts.xmp, makes no difference) which contains a tag DateCreated with the correct timestamp, i.e.: `photoshop:DateCreated="2009-06-24T12:21:26+03:00"`
- When I import the video file with Photoprism, it insists on indexing it based on the 2007:01:01 timestamp.

So what exactly am I doing wrong? :)

**Reply by lastzero** [reply]
As import was built to quickly import files eg from SD cards as provided by digital cameras, it may not include metadata from XMP sidecar files when calculating the creation date (yet) as those are typically created, for example by Photoshop, after images have been initially imported.

**Reply by Nocifer** [reply]
Ah, alright. So if I'm not mistaken, my only option for the time being is to manually copy over the (thankfully few) files that contain problematic metadata into the 'originals' hierarchy and then edit them with the correct datetime information from within Photoprism, storing their metadata in the YAML sidecars. And when Photoprism matures some more, it will hopefully gain the ability both to fully read EXIF/XMP metadata and further populate its YAML files accordingly, and to write metadata back into its own XMP files and/or directly to the media files using the information I'll now be editing manually into the YAML files.

Thanks for taking the time to reply to my questions, and keep up the good work!

**Reply by lastzero** [reply]
I think JSON might work instead. Any reason you want to use XMP?

**Reply by Nocifer** [reply]
No reason at all. It's just that I'd also tried with JSON before I came here to ask my original question (using `exiftool -json $FILE`  which produces output like e.g. `aurora.jpg.json` from that list of examples you've posted) but Photoprism still imported the file with the wrong timestamp, so at that point I kind of gave up. But since you're implying that JSON should actually be the solution I'm looking for, I guess it might be me who didn't do things properly in this case (I didn't test extensively with JSON) so I'll try again and see where that gets me.

**Reply by Nocifer** [reply]
After testing again, I found that using a JSON file with the correct `DateTimeOriginal` timestamp resulted in the files being imported correctly into the `originals` folder hierarchy, but their metadata (i.e. the resulting YAML sidecar) still contained a wrong `TakenAt` timestamp, presumably because the file's EXIF metadata (or possibly the file creation date) still take priority over the JSON file's values when it comes to the creation of the YAML sidecar.

So unless I'm doing something wrong, it seems that for the time being I still need to do some manual adjustment after importing.

**Reply by lastzero** [reply]
Metadata sources have different priorities for PhotoPrism, depending on the format and where the information comes from:

https://github.com/photoprism/photoprism/blob/a7872a684b78d3b80baff977ea0615c1eccb731f/internal/entity/src.go#L9-L20

One way to override `TakenAt` would be to use YAML files, since you can specify the data source there:

https://github.com/photoprism/photoprism/blob/a7872a684b78d3b80baff977ea0615c1eccb731f/internal/entity/photo.go#L53

Based on the above overview, the value you should specify in `TakenSrc` is `manual` since no other data source has a higher priority.

You can check the YAML files in the sidecar folder to see examples if you have not disabled the creation of these files in your local configuration. Hope this helps!
